Brokerages/Alpaca/AlpacaBrokerage.Utility.cs

```diff
@@ -0,0 +1,428 @@
+            Log.Error(exception, "SockClient error");
+        }
+
+        /// <summary>
+        /// Downloads a list of TradeBars at the requested resolution
+        /// </summary>
+        /// <param name="symbol">The symbol</param>
+        /// <param name="startTimeUtc">The starting time (UTC)</param>
+        /// <param name="endTimeUtc">The ending time (UTC)</param>
+        /// <param name="resolution">The requested resolution</param>
+        /// <param name="requestedTimeZone">The requested timezone for the data</param>
+        /// <returns>The list of bars</returns>
+        private IEnumerable<TradeBar> DownloadTradeBars(Symbol symbol, DateTime startTimeUtc, DateTime endTimeUtc, Resolution resolution, DateTimeZone requestedTimeZone)
+        {
+            // Only minute/hour/daily resolutions supported
+            if (resolution < Resolution.Minute)
+            {
+                yield break;
+            }
+
+            var period = resolution.ToTimeSpan();
+
+            var startTime = startTimeUtc.RoundDown(period);
+            var endTime = endTimeUtc.RoundDown(period).Add(period);
+
+            while (startTime < endTime)
+            {
+                CheckRateLimiting();
+
+                var task = resolution == Resolution.Daily
+                    ? _restClient.ListDayAggregatesAsync(symbol.Value, startTime, endTime)
+                    : _restClient.ListMinuteAggregatesAsync(symbol.Value, startTime, endTime);
+
+                var time = startTime;
+                var items = task.SynchronouslyAwaitTaskResult()
+                    .Items
+                    .Where(x => x.Time >= time)
+                    .ToList();
+
+                if (!items.Any())
+                {
+                    break;
+                }
+
+                if (resolution == Resolution.Hour)
+                {
+                    // aggregate minute tradebars into hourly tradebars
+                    var bars = items
+                        .GroupBy(x => x.Time.RoundDown(period))
+                        .Select(
+                            x => new TradeBar(
+                                x.Key.ConvertFromUtc(requestedTimeZone),
+                                symbol,
+                                x.First().Open,
+                                x.Max(t => t.High),
+                                x.Min(t => t.Low),
+                                x.Last().Close,
+                                x.Sum(t => t.Volume),
+                                period
+                            ));
+
+                    foreach (var bar in bars)
+                    {
+                        yield return bar;
+                    }
+                }
+                else
+                {
+                    foreach (var item in items)
+                    {
+                        yield return new TradeBar(
+                            resolution == Resolution.Daily
+                                ? item.Time
```

Comment:
Why is the `time` not converted to the `requestedTimeZone` for `Resolution.Daily`?